pkg/ipam/node.go

```diff
@@ -653,6 +658,15 @@ func (n *Node) updateLastResync(syncTime time.Time) {
 // MaintainIPPool attempts to allocate or release all required IPs to fulfill
 // the needed gap. If required, interfaces are created.
 func (n *Node) MaintainIPPool(ctx context.Context) error {
+	// As long as the instances API is unstable, don't perform any
+	// operation that can mutate state.
+	if !n.manager.InstancesAPIIsReady() {
+		if n.retry != nil {
+			n.retry.Trigger()
+		}
+		return fmt.Errorf("instances API is unstable. Blocking mutating operations. See logs for details.")
```

Comment:
Nit: we can use `errors.New()` for simple string errors. This can help reduce the usage and hopefully the need to import `fmt`, but I assume that we already use `fmt.Errorf` for good reasons in this file. Feel free to ignore if so.

Reply:
I'm keeping `fmt.Errorf()` for now to keep it consistent with the rest of the code in this file. I think we can definitely start using `errors.New()` and `errors.Wrap()` but we should make a consistent decision to do so after some discussion because I'm sure it will have implications on assumptions that all contributors are currently making.